mail_outbound_static/tests/test_ir_mail_server.py

```diff
@@ -0,0 +1,384 @@
+        used_mail_server = self.IrMailServer.browse(used_mail_server)
+        self.assertEqual(
+            used_mail_server,
+            expected_mail_server,
+            (
+                f"It using {used_mail_server.name}"
+                f" but we expect to use {expected_mail_server.name}"
+            ),
+        )
+
+    def test_02_from_outgoing_server_domaintwo(self):
+        self._init_mail_server_domain_whilelist_based()
+        domain = "domaintwo.com"
+        email_from = f"Mitchell Admin <admin@{domain}>"
+        expected_mail_server = self.mail_server_domaintwo
+
+        self.message.replace_header("From", email_from)
+        # A mail server is configured for the email
+        with self.mock_smtplib_connection():
+            message = self._send_mail(self.message)
+        self.assertEqual(message["From"], email_from)
+
+        used_mail_server = self.IrMailServer._get_mail_sever(domain)
+        used_mail_server = self.IrMailServer.browse(used_mail_server)
+        self.assertEqual(
+            used_mail_server,
+            expected_mail_server,
+            (
+                f"It using {used_mail_server.name}"
+                f" but we expect to use {expected_mail_server.name}"
+            ),
+        )
+
+    def test_03_from_outgoing_server_another(self):
+        self._init_mail_server_domain_whilelist_based()
+        domain = "example.com"
+        email_from = f"Mitchell Admin <admin@{domain}>"
+        expected_mail_server = self.mail_server_domainone
+
+        self.message.replace_header("From", email_from)
+        # A mail server is configured for the email
+        with self.mock_smtplib_connection():
+            message = self._send_mail(self.message)
+        self.assertEqual(
+            message["From"], f"Mitchell Admin <{expected_mail_server.smtp_from}>"
+        )
+
+        used_mail_server = self.IrMailServer._get_mail_sever(domain)
+        used_mail_server = self.IrMailServer.browse(used_mail_server)
+        self.assertEqual(
+            used_mail_server,
+            expected_mail_server,
+            (
+                f"It using {used_mail_server.name}"
+                f" but we expect to use {expected_mail_server.name}"
+            ),
+        )
+
+    @patch.dict(tools.config.options, {"smtp_from": "admin@example.com"})
+    @patch.dict(tools.config.options, {"smtp_domain_whitelist": "example.com"})
+    def test_04_from_outgoing_server_none_use_config(self):
+        self._init_mail_server_domain_whilelist_based()
+        domain = "example.com"
+        email_from = f"Mitchell Admin <admin@{domain}>"
+
+        self._delete_mail_servers()
+        self.assertFalse(self.IrMailServer.search([]))
+        # Find config values
+        config_smtp_from = tools.config.get("smtp_from")
+        config_smtp_domain_whitelist = tools.config.get("smtp_domain_whitelist")
+        if not config_smtp_from or not config_smtp_domain_whitelist:
+            self._skip_test(
```

Comment:
this and skip_test above can go away too with the above, right?

Reply:
@hbrunn Right! Thank you!